pkg/sdkserver/sdkserver.go

```diff
@@ -985,41 +999,238 @@ func (s *SDKServer) GetList(ctx context.Context, in *alpha.GetListRequest) (*alp
+		// TODO: This does not check batched but not yet applied append / remove values. Should we do this?
+		// (Easy to check not yet applied values, hard to check removed and re-added values.) I'm
+		// thinking this would be better / easier to do as part of the batch apply update to list, and
+		// not verify here.
+		for _, val := range list.Values {
+			if in.Value == val {
+				return nil, errors.Errorf("ALREADY_EXISTS. Value: %s already in List: %s", in.Value, in.Name)
+			}
+		}
+		batchList.listAppend = append(batchList.listAppend, in.Value)
+		s.gsListUpdates[in.Name] = batchList
+		// Queue up the Update for later batch processing by updateLists.
+		s.workerqueue.Enqueue(cache.ExplicitKey(updateLists))
+		return &alpha.List{}, nil
+	}
+	return nil, errors.Errorf("NOT_FOUND. %s List not found", in.Name)
 }
 
-// RemoveListValue returns the updated List.
+// RemoveListValue collapses all remove a value from a List requests into a single UpdateList request.
+// Returns NOT_FOUND if the List does not exist.
+// Returns NOT_FOUND if the value is not in the List.
 // [Stage:Alpha]
 // [FeatureFlag:CountsAndLists]
 func (s *SDKServer) RemoveListValue(ctx context.Context, in *alpha.RemoveListValueRequest) (*alpha.List, error) {
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement Me
-	return nil, errors.Errorf("Unimplemented -- RemoveListValue coming soon")
+
+	s.logger.WithField("name", in.Name).Debug("Remove List Value")
+
+	s.gsUpdateMutex.Lock()
+	defer s.gsUpdateMutex.Unlock()
+
+	gs, err := s.gameServer()
+	if err != nil {
+		return nil, err
+	}
+
+	if list, ok := gs.Status.Lists[in.Name]; ok {
+		// Verify value exists in the list
+		for _, val := range list.Values {
+			if in.Value != val {
+				continue
+			}
+			// Add value to remove to gsListUpdates map.
+			batchList := s.gsListUpdates[in.Name]
+			batchList.listDelete = append(batchList.listDelete, in.Value)
+			s.gsListUpdates[in.Name] = batchList
+			// Queue up the Update for later batch processing by updateLists.
+			s.workerqueue.Enqueue(cache.ExplicitKey(updateLists))
+			return &alpha.List{}, nil
+		}
+		return nil, errors.Errorf("NOT_FOUND. Value: %s not found in List: %s", in.Value, in.Name)
+	}
+	return nil, errors.Errorf("NOT_FOUND. %s List not found", in.Name)
+}
+
+// updateList updates the Lists in the GameServer's Status with the batched update list requests.
+// Includes all SetCapacity, AddValue, and RemoveValue requests in the batched request.
+func (s *SDKServer) updateList(ctx context.Context) error {
+	gs, err := s.gameServer()
+	if err != nil {
+		return err
+	}
+	gsCopy := gs.DeepCopy()
+
+	s.logger.WithField("batchListUpdates", s.gsListUpdates).Debug("Batch updating List(s)")
+	s.gsUpdateMutex.Lock()
+	defer s.gsUpdateMutex.Unlock()
+
+	names := []string{}
+
+	for name, listReq := range s.gsListUpdates {
```

Comment:
As a potetially interesting thought - I would figure `GetList()` would do this work as well / already.

You could port that into there, and call `GetList(...)` from here as well to get what should be the source of truth and then update?

Reply:
I think I tried that with Counter and it was complicated with the s.gsUpdateMutex.Lock(). Any idea how to do that?

Reply:
The way I would do it is move the implementation of `GetList` into a non locked private function, and make sure to document that calls to that function must have appropriate locking.

Then `GetList` and all the other functions can call that private function to get the implementation, but can have control over the over arching mutex lock before the invocation of the non-locking private function.

Reply:
Looking at this again, I'm not sure we want to use GetList here. Here we get the gameserver (and lock on it), and update the values directly in the gameserver copy. If we use "Get List" we'd have to do that for each list, which would be calling s.gameServer() for each list.

Reply:
They're still somewhat different. This method updates agonesv1.ListStatus and the Getlist updates alpha.List.